astoria/managers/mixins/disk_handler.py

```diff
@@ -0,0 +1,57 @@
+"""Disk Handler Mixin."""
+
+import asyncio
+import logging
+from json import JSONDecodeError, loads
+from typing import Dict, Match
+
+from astoria.common.messages.astdiskd import (
+    DiskInfo,
+    DiskManagerMessage,
+    DiskUUID,
+)
+
+LOGGER = logging.getLogger(__name__)
+
+
+class DiskHandlerMixin:
+    """Mixin to translate disk events into insertions and removals."""
+
+    _cur_disks: Dict[DiskUUID, DiskInfo]
+
+    async def handle_astdiskd_disk_info_message(
+        self,
+        match: Match[str],
+        payload: str,
+    ) -> None:
+        """Handle disk info messages."""
+        if payload:
+            try:
+                message = DiskManagerMessage(**loads(payload))
+
+                new_set = set(message.disks.keys())
+                old_set = set(self._cur_disks.keys())
+
+                added_disks = new_set - old_set
+                removed_disks = old_set - new_set
+
+                for uuid in removed_disks:
+                    info = self._cur_disks.pop(uuid)
+                    asyncio.ensure_future(self.handle_disk_removal(uuid, info))
```

Comment:
will this (and the one below) result in warnings about futures not being awaited?

Reply:
The event loop shouldn't finish until all tasks (technically futures) have been completed, which will include these ones. 
`ensure_future` is equivalent to `asyncio.create_task` (`create_task` is 3.7+ only)

When we drop support for 3.6, we should change all of these to `create_task` to make the code more readable.